app/src/main/java/de/danoeh/antennapod/dialog/FilterDialog.java

```diff
@@ -21,42 +29,143 @@ public FilterDialog(Context context, FeedItemFilter feedItemFilter) {
         this.filter = feedItemFilter;
     }
 
-    public void openDialog() {
-        final String[] items = context.getResources().getStringArray(R.array.episode_filter_options);
-        final String[] values = context.getResources().getStringArray(R.array.episode_filter_values);
-        final boolean[] checkedItems = new boolean[items.length];
+    public void openCustomDialog() {
 
         final Set<String> filterValues = new HashSet<>(Arrays.asList(filter.getValues()));
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle(R.string.filter);
 
-        // make sure we have no empty strings in the filter list
-        for (String filterValue : filterValues) {
-            if (TextUtils.isEmpty(filterValue)) {
-                filterValues.remove(filterValue);
-            }
-        }
+        LayoutInflater inflater = LayoutInflater.from(this.context);
+        View customLayout = inflater.inflate(R.layout.custom_filter_dialog_layout, null);
+        TableLayout tableLayout = customLayout.findViewById(R.id.custom_dialog_tableLayout1);
+
+        builder.setView(customLayout);
+
+        for (FeedItemEnum item : FeedItemEnum.values()) {
+
+            RelativeLayout row = (RelativeLayout) inflater.inflate(R.layout.custom_dialog_filter_row, null);
+            TextView textView1 = row.findViewById(R.id.custom_dialog_textView1);
+            TextView textView2 = row.findViewById(R.id.custom_dialog_textView2);
+            TextView textView3 = row.findViewById(R.id.custom_dialog_textView3);
```

Comment:
This is now fixed with 3 TextViews. My idea was to keep it independent from the number of options. So the views should be inflated one by one from another layout, just as needed. This should also make the code less verbose because you do not have to write everything multiple times (instead, using loops).